test/README.md

```diff
@@ -0,0 +1,49 @@
+# How to run the end-to-end tests
+
+## Test parameters
+
+The end-to-end tests take several parameters, which they expect to find in a file named `context.json` in the same directory as the test binaries (e.g., `test/Microsoft.Azure.Devices.Edge.Test/bin/Debug/netcoreapp2.1/context.json`). Parameter names are case-insensitive. The parameters are:
+
+| Name | Required | Description |
+|------|----------|-------------|
+| `caCertScriptPath` | ✔ | Path to the folder containing `certGen.sh` (Linux) or `ca-certs.ps1` (Windows). |
+| `deviceId` || Name by which the edge device-under-test will be registered in IoT Hub. If not given, a name will automatically be generated in the format: `e2e-{device hostname}-{yyMMdd-HHmmss.fff}`. |
+| `edgeAgentImage` || Docker image to pull/use for Edge Agent. Regardless of what you put here, the default `mcr.microsoft.com/azureiotedge-agent:1.0` will be used in config.yaml to start IoT Edge; this setting only applies to any configurations deployed by the tests. |
+| `edgeHubImage` || Docker image to pull/use for Edge Hub. |
```

Comment:
does it has default image like edge agent?

Reply:
No you have to specify it in context.json.

Reply:
BTW Edge Agent has a default in config.yaml, but like Edge Agent it does NOT have a default in the deployment. You have to specify it in context.json.